### PR TITLE
Rebuilt animal api. Now easier to drop in other web apis for animals.…

### PR DESCRIPTION
**Pull request by Quesos-Nachos**
… Removed cat module and fox module in favor of animalmodule. Changed some fox module http code.

### CODE REVIEW

File: Modix/Services/Animals/AnimalService.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                new GiraffeDucksApi()
+            };
+
+            /*_dogApis = new List<IAnimalApi>
+            {
+
+            };*/
+        }
+
+        public async Task<Response> Get(AnimalType animalType, MediaType mediaType, CancellationToken cancellationToken = default)
```
Comment (mastorm) [inline Modix/Services/Animals/AnimalService.cs]: *GetAsync

File: Modix/Services/Animals/AnimalService.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public interface IAnimalService
+    {
+        /// <summary>
+        /// Gets an image
+        /// </summary>
+        /// <param name="animalType">The type of animal we want to see</param>
+        /// <param name="mediaType">The type of media we want: an animated gif or a still picture</param>
+        /// <param name="cancellationToken">The cancellation token for our time limit</param>
+        /// <returns></returns>
+        Task<Response> Get(AnimalType animalType, MediaType mediaType, CancellationToken cancellationToken = default);
```
Comment (mastorm) [inline Modix/Services/Animals/AnimalService.cs]: *GetAsync

File: Modix/Modules/AnimalModule.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                var reply = await _animalService.Get(animal, mediaType);
+
+                if (reply != null)
+                {
+                    await ProcessResponse(mediaType, reply);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                await ReplyAsync("Couldn't get a fox picture in time :(");
```
Comment (sylveon) [inline Modix/Modules/AnimalModule.cs]: Change this to use the animal name from the AnimalType parameter

File: Modix/Modules/AnimalModule.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            }
+            catch (Exception exc)
+            {
+                await ReplyAsync($"Couldn't get a fox picture: {exc.Message}");
```
Comment (sylveon) [inline Modix/Modules/AnimalModule.cs]: Same

**Pull request merged**